### PR TITLE
fix: validate proto messages before converting them to anypb.Any

### PR DESCRIPTION
**Pull request by zhaohuabing**
The `validate` and `validateAll` method of a proto message don't validate a member field of `anypb.Any` type as it doesn't have a `validate` method itself.  Therefore, all proto messages must be validated before been converting to `anypb.Any`.

Fix https://github.com/envoyproxy/gateway/issues/4504

Release note: No

### CODE REVIEW

File: internal/xds/translator/testdata/in/xds-ir/accesslog.yaml [hunk 1]
```diff
@@ -13,7 +13,8 @@ accesslog:
       protocol: "%PROTOCOL%"
       response_code: "%RESPONSE_CODE%"
   als:
-  - destination:
+  - name: als
```
Comment (zhaohuabing) [inline internal/xds/translator/testdata/in/xds-ir/accesslog.yaml:16]:
```
--- FAIL: TestTranslateXds (0.00s)
    --- FAIL: TestTranslateXds/accesslog (0.00s)
        translator_test.go:142: accesslog
        translator_test.go:143: 
                Error Trace:    /home/ubuntu/gateway/internal/xds/translator/translator_test.go:143
                Error:          Received unexpected error:
                                invalid HttpGrpcAccessLogConfig.CommonConfig: embedded message failed validation | caused by: invalid CommonGrpcAccessLogConfig.LogName: value length must be at least 1 runes
                Test:           TestTranslateXds/accesslog
```

File: internal/xds/translator/testdata/in/xds-ir/accesslog-without-format.yaml [hunk 2]
```diff
@@ -11,7 +11,8 @@ accesslog:
       protocol: "%PROTOCOL%"
       response_code: "%RESPONSE_CODE%"
   als:
-  - destination:
+  - name: als
```
Comment (zhaohuabing) [inline internal/xds/translator/testdata/in/xds-ir/accesslog-without-format.yaml:14]:
```
go test -timeout 30s github.com/envoyproxy/gateway/internal/xds/translator --override-testdata=true
--- FAIL: TestTranslateXds (0.42s)
    --- FAIL: TestTranslateXds/accesslog-without-format (0.00s)
        translator_test.go:142: accesslog-without-format
        translator_test.go:143: 
                Error Trace:    /home/ubuntu/gateway/internal/xds/translator/translator_test.go:143
                Error:          Received unexpected error:
                                invalid HttpGrpcAccessLogConfig.CommonConfig: embedded message failed validation | caused by: invalid CommonGrpcAccessLogConfig.LogName: value length must be at least 1 runes
```

File: internal/xds/translator/testdata/in/xds-ir/authorization-multiple-principals.yaml [hunk 3]
```diff
@@ -44,7 +44,7 @@ http:
               isIPv6: false
               maskLen: 24
             jwt:
-              issuer: https://one.example.com
+              provider: https://one.example.com
```
Comment (zhaohuabing) [inline internal/xds/translator/testdata/in/xds-ir/authorization-multiple-principals.yaml:47]:
```
--- FAIL: TestTranslateXds (0.00s)
    --- FAIL: TestTranslateXds/authorization-multiple-principals (0.00s)
        translator_test.go:142: authorization-multiple-principals
        translator_test.go:143: 
                Error Trace:    /home/ubuntu/gateway/internal/xds/translator/translator_test.go:143
                Error:          Received unexpected error:
                                invalid DynamicMetadataInput.Path[0]: embedded message failed validation | caused by: invalid DynamicMetadataInput_PathSegment.Key: value length must be at least 1 runes
                Test:           TestTranslateXds/authorization-multiple-principals
```

File: internal/utils/protocov/protocov.go [hunk 4]
```diff
@@ -12,30 +12,30 @@ import (
...
-const (
-	APIPrefix = "type.googleapis.com/"
-)
-
-var marshalOpts = proto.MarshalOptions{}
+// Deprecated: error should not be ignored, use ToAnyWithValidation instead.
+func ToAny(msg proto.Message) *anypb.Any {
+	res, err := ToAnyWithValidation(msg)
+	if err != nil {
+		return nil
```
Comment (zhaohuabing) [inline internal/utils/protocov/protocov.go:19]: The validation error should not be ignored. 

Deprecated the `ToAny` function and will clean up all the refrences to it in a follow-up PR.

Comment (zirain) [reply internal/utils/protocov/protocov.go:19]: for case like following, ToAny still meanfull.
```console
GRPCWeb = &hcm.HttpFilter{
		Name: wellknown.GRPCWeb,
		ConfigType: &hcm.HttpFilter_TypedConfig{
			TypedConfig: protocov.ToAny(&grpcweb.GrpcWeb{}),
		},
	}
```

**Review by zirain — DISMISSED** [review]

File: internal/utils/protocov/protocov.go (hunk 4 above)
Comment (zirain) [reply internal/utils/protocov/protocov.go:19]: should we change the comment?

**Review by zirain — APPROVED** [review]

**Comment by zirain** [comment]
this's worth to backport if possible.

File: internal/utils/protocov/protocov.go (hunk 4 above)
Comment (zhaohuabing) [reply internal/utils/protocov/protocov.go:19]: I prefer removing this function at all, but let's discuss and do it in a follow-up PR.

**Review by arkodg — APPROVED** [review]
LGTM thanks for fixing it across the board

**Pull request merged**